mrp_lot_number_propagation/models/mrp_production.py

```diff
@@ -92,9 +91,112 @@ def _set_lot_number_propagation_data_from_bom(self):
+            # Check if all batch productions are set to propagate lot number
+            productions_auto_ids = set()
+            productions_batch_ids = set()
+            for order in self:
+                if order._auto_production_checks():
+                    productions_auto_ids.add(order.id)
+                else:
+                    productions_batch_ids.add(order.id)
+
+            batch_productions = self.browse(productions_batch_ids)
+            for order in batch_productions:
+                # If any production is not set to propagate, it will require user input
+                #  so return the wizard for that order instead of one with lot number
+                #  propagation
+                if not order.is_lot_number_propagated:
+                    res["context"] = {"default_production_id": order.id}
+                    return res
+                # If any production is set to propagate, but has multiple tracked
+                #  components, we cannot mass produce without user input for the
+                #  components that don't propagate their lot number.
+                #  so return the wizard for that order
+                if (
+                    len(
+                        order.move_raw_ids.filtered(
+                            lambda mv: mv.product_id.tracking != "none"
+                        )
+                    )
+                    > 1
+                ):
+                    res["context"] = {"default_production_id": order.id}
+                    return res
+            # If we are here, it means we must be able to produce everything
+            #  automatically
+            # Now we need to finish whatever checks were meant to be done through
+            #  pre_button_mark_done, but weren't finished as the batch product
+            #  wizard had to be returned
+            # FIXME: If an order to be propagated has consumption issue, and is meant
+            #  to create backorder this will call button_mark_done, without having set
+            #  up lot propagation beforehand
+            res = super(
+                MrpProduction,
+                self.with_context(bypass_pre_button_mark_done_checks=True),
+            ).pre_button_mark_done()
+            # If any other wizard action had to be returned, (eg consumption issue,
+            #  backorder) let's return it
+            if isinstance(res, dict):
+                return res
+            # Now we can safely handle productions set to propagate lot number
+            for order in batch_productions:
+                batch_produce_wizard = (
+                    self.env["mrp.batch.produce"]
+                    .with_context(default_production_id=order.id)
+                    .create({})
+                )
+                propagate_move = order._get_propagating_component_move()
+                batch_produce_wizard.production_text = "\n".join(
+                    [
+                        f"{ml.lot_id.name},{ml.lot_id.name}"
```

Comment:
I don't know this `production_text` field, but here you should not use `self.component_separator` instead of comma, as you are splitting on this in the `mrp.batch.produce` wizard?